itest/starboard/starboard_cli_test.go

```diff
@@ -61,6 +64,63 @@ func NewPodSpec(podName string, containers map[string]string, args ...string) (*
 	return pod, err
 }
 
+func IsVulnerabilityReportForContainerOwnedBy(containerName string, owner client.Object) types.GomegaMatcher {
+	return &vulnerabilityReportMatcher{
+		containerName: containerName,
+		owner:         owner,
+	}
+}
+
+type vulnerabilityReportMatcher struct {
+	owner         client.Object
+	containerName string
+
+	failureMessage        string
+	negatedFailureMessage string
+}
+
+func (matcher *vulnerabilityReportMatcher) Match(actual interface{}) (success bool, err error) {
+	_, ok := actual.(v1alpha1.VulnerabilityReport)
+	if !ok {
+		return false, fmt.Errorf("vulnerabilityReportMatcher expects a VulnerabilityReport")
+	}
+	gvk, err := apiutil.GVKForObject(matcher.owner, starboard.NewScheme())
+	if err != nil {
+		return false, err
+	}
+	m := MatchFields(IgnoreExtras, Fields{
+		"ObjectMeta": MatchFields(IgnoreExtras, Fields{
+			"Labels": MatchAllKeys(Keys{
+				kube.LabelContainerName:     Equal(matcher.containerName),
+				kube.LabelResourceKind:      Equal(gvk.Kind),
+				kube.LabelResourceName:      Equal(matcher.owner.GetName()),
+				kube.LabelResourceNamespace: Equal(matcher.owner.GetNamespace()),
+			}),
+			"OwnerReferences": ConsistOf(metav1.OwnerReference{
+				APIVersion: gvk.GroupVersion().Identifier(),
+				Kind:       gvk.Kind,
+				Name:       matcher.owner.GetName(),
+				UID:        matcher.owner.GetUID(),
+			}),
+		}),
+		"Report": MatchFields(IgnoreExtras, Fields{
+			"Scanner": Equal(trivyScanner),
+		}),
+	})
+	matcher.failureMessage = m.FailureMessage(actual)
+	matcher.negatedFailureMessage = m.NegatedFailureMessage(actual)
+
+	return m.Match(actual)
+}
+
+func (matcher *vulnerabilityReportMatcher) FailureMessage(actual interface{}) (message string) {
+	return matcher.failureMessage
+}
+
+func (matcher *vulnerabilityReportMatcher) NegatedFailureMessage(actual interface{}) (message string) {
+	return matcher.negatedFailureMessage
+}
```

Comment:
Just for my understanding: Where are the values for these fields initialized?

Reply:
A custom matcher is stateful and we set those values in the `Match(actual interface{})` callback. Lines 110-111.

Reply:
However, I realised that I assigned messages before calling the `Match` method 🤦 I've already fixed that, but in general it is guaranteed that `FailureMessage` and `NegatedFailureMessage` will only be called after `Match`, so you can save off any state you need to compute the messages in `Match`. See https://onsi.github.io/gomega/#adding-your-own-matchers

Reply:
I see that makes a lot more sense. Cool!